### ISSUE TITLE
Unexpected token when dependency uses `#` to indicate private method during `mm-build`

### ISSUE THREAD

**Issue by alanorwick** [MetaMask/snaps-monorepo#1367]
Seeing an issue when using `ethers@6.3.0` during `mm-build` that results from `abi-coder.ts` having `#getCoder`.

```
Build error: Parsing file /Users/user/quai-snap/node_modules/ethers/lib.commonjs/abi/abi-coder.js: Unexpected token (112:4)
Error: Parsing file /Users/user/quai-snap/node_modules/ethers/lib.commonjs/abi/abi-coder.js: Unexpected token (112:4)
    at Deps.parseDeps (/Users/user/quai-snap/node_modules/module-deps/index.js:519:15)
    at getDeps (/Users/user/quai-snap/node_modules/module-deps/index.js:447:44)
    at /Users/user/quai-snap/node_modules/module-deps/index.js:430:38
    at ConcatStream.<anonymous> (/Users/user/quai-snap/node_modules/concat-stream/index.js:37:43)
    at ConcatStream.emit (node:events:525:35)
    at finishMaybe (/Users/user/quai-snap/node_modules/readable-stream/lib/_stream_writable.js:630:14)
    at endWritable (/Users/user/quai-snap/node_modules/readable-stream/lib/_stream_writable.js:638:3)
    at Writable.end (/Users/user/quai-snap/node_modules/readable-stream/lib/_stream_writable.js:594:22)
    at DuplexWrapper.onend (/Users/user/quai-snap/node_modules/readable-stream/lib/_stream_readable.js:577:10)
    at Object.onceWrapper (node:events:627:28)
    at DuplexWrapper.emit (node:events:525:35)
    at endReadableNT (/Users/user/quai-snap/node_modules/readable-stream/lib/_stream_readable.js:1010:12)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21) {

**Comment by FrederikBolding** [comment]
Looking into this. Looks like our Babel configuration is not compatible with the latest version of Ethers.

**Comment by alanorwick** [comment]
> I am working on a fix for this in PR #1373 - should be merged soon and fixed in a future release. If you need this fixed urgently you can patch the snaps CLI with the updates from my PR and enable transpiling of dependencies by changing the `transpilationMode` in `snap.config.js` to `'localAndDeps'`

Good to know, will track this for the future. Thank you!
